src/access/pg_tdeam.c

```diff
@@ -3112,9 +3114,24 @@ pg_tde_update(Relation relation, ItemPointer otid, HeapTuple newtup,
 	oldtup.t_len = ItemIdGetLength(lp);
 	oldtup.t_self = *otid;
 	/* decrypt the old tuple */
-	PG_TDE_DECRYPT_TUPLE(BufferGetBlockNumber(buffer), page, &oldtup, &oldtup,
+	{
+		MemoryContext oldContext;
+		char* new_ptr = NULL;
+		oldContext = MemoryContextSwitchTo(CurTransactionContext);
+		new_ptr = palloc(oldtup.t_len);
+		MemoryContextSwitchTo(oldContext);
```

Comment:
```suggestion
		char* new_ptr = NULL;
		new_ptr = MemoryContextAlloc(CurTransactionContext, oldtup.t_len);
```
Really nitpicking but to me its looks cleaner while doing the same job.